posit-dev/positron
# Minimal notebook UI flag for vscode notebooks

This PR adds a new mode enabled by the feature flag `notebook.usePositronMinimalNotebookUI`
<img width="919" alt="image" src="https://github.com/user-attachments/assets/bb4a625c-01d6-47db-89f0-f134bfd241f2">

The flag defaults to off. 

## Static changes

There are a few static visual changes:

![image](https://github.com/user-attachments/assets/b821d142-f08c-4d77-beed-0c45d01c1dd2)


## Animation changes
Perhaps more important are the changes to the experience of navigating and running cells in the notebook.
Previously when you would run a cell the following animations would activate
- Spinner in the  main notebook toolbar
- A _different_ spinner beneath the cell
- The cell execution-time counter
- A bouncing progress bar above the cell. 

Minimal mode removes the cell-level spinner and the bouncing progress bar to keep things a bit less hectic. 

### Running a cell with defaults

https://github.com/user-attachments/assets/82efe13a-d311-4596-82cd-fbd4592f5e29



### Running cells with minimal mode
https://github.com/user-attachments/assets/746d64bb-91ed-46c4-bc96-6569900f49e3




### QA Notes

This PR is almost entirely implemented via css, so there shouldn't be too many hiccups in terms of things changing. Since it's purely UI-based I held off on adding a smoke-test as that felt a bit overboard for this type of change. 
Basically, after changing the feature flag, you should be able to open up any notebook you could open before and have things look a bit more... calm.
